src/rpc/net.cpp

```diff
@@ -146,7 +146,7 @@ static RPCHelpMan getpeerinfo()
                     {
                         {RPCResult::Type::STR, "permission_type", Join(NET_PERMISSIONS_DOC, ",\n") + ".\n"},
                     }},
-                    {RPCResult::Type::NUM, "minfeefilter", /*optional=*/true, "The minimum fee rate for transactions this peer accepts"},
```

Comment:
in 37d5cdbaac6f84aff90942c03f83e094a09c7bde:

I see this is a partial revert of #24691, which was a follow up to #21160? This commit would be easier to follow without the previous move-only commit.

> This was inadvertently changed on master after the v23 branch-off.

It would be good to link to actual discussion / history rather than saying "inadvertently changed". It's not clear that anything inadvertent happened. The change was made, docs/optional was added / fixed, but now it's being changed back again.

Reply:
See the discussion from https://github.com/bitcoin/bitcoin/pull/26109#discussion_r973104660. It's hard to tell what happened and inadvertent seems the kindest explanation if one is needed, along with not linking.  If any fields become newly optional in v24, ISTM that should warned about in a release note (that users can be pointed to, if they ask what happened).